snuba/datasets/configuration/entity_builder.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+import logging
+from typing import Any, Mapping, Sequence, Type
+
+from snuba.clickhouse.translators.snuba.allowed import (
+    FunctionCallMapper,
+    SubscriptableReferenceMapper,
+)
+from snuba.clickhouse.translators.snuba.mappers import (
+    FunctionNameMapper,
+    SubscriptableMapper,
+)
+from snuba.clickhouse.translators.snuba.mapping import TranslationMappers
+from snuba.datasets.configuration.json_schema import V1_ENTITY_SCHEMA
+from snuba.datasets.configuration.loader import load_configuration_data
+from snuba.datasets.configuration.utils import parse_columns
+from snuba.datasets.entity import Entity
+from snuba.datasets.pluggable_entity import PluggableEntity
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_storage, get_writable_storage
+from snuba.query.processors import QueryProcessor
+from snuba.query.validation.validators import (
+    EntityRequiredColumnValidator,
+    QueryValidator,
+)
+
+__MAPPINGS_HAVE_BEEN_INITIALIZED = False
```

Comment:
Why have a variable for this? Could you not check the length of mappings?

Reply:
happy to change to a length check